ocs_ci/deployment/hub_spoke.py

```
@@ -3066,13 +3157,43 @@ def _reconstruct_infra_output(self, existing_vpcs):
-                f"This is a leftover from a previous failed run. "
-                f"Delete the VPC and its associated resources, then retry."
+            mgmt_tag = self._get_mgmt_cluster_tag_from_vpc(existing_vpcs[0])
+            our_mgmt = config.ENV_DATA.get("cluster_name", "")
+
+            if not mgmt_tag:
+                raise RuntimeError(
+                    f"Partial infrastructure detected for cluster '{self.name}': "
+                    f"VPC '{vpc_id}' exists but Route53 zones are missing and the "
+                    f"VPC has no '{self.MGMT_CLUSTER_TAG_KEY}' tag. Cannot determine "
+                    f"ownership — manual cleanup required."
+                )
```

Ruff (0.16.0)

[warning] Avoid specifying long messages outside the exception class

(TRY003)